litgpt/data/tinystories.py

```diff
@@ -27,155 +24,117 @@ class TinyStories(DataModule):
     Provides training and validation dataloaders that return batches of tokens. Every sample is set to a fixed length.
     """
 
-    path: Path = Path("data/")
-    """Path to the data directory where data will be downloaded and preprocessed"""
-    num_workers: int = 0
-    """How many DataLoader processes to use for loading."""
+    data_path: Path = Path("data/tinystories")
+    """The path to the data directory, containing two folders 'train' and 'val'
+    which are the output of the preprocessing step. The path can also be a remote path (e.g., s3://)."""
     seed: int = 42
-    """The random seed for creating the train/val splits and shuffling the dataset."""
+    """The seed to use for shuffling the dataset."""
+    num_workers: int = 8
+    """The number of workers to use for the dataloaders."""
 
     tokenizer: Optional[Tokenizer] = field(default=None, init=False, repr=False)
     batch_size: int = field(default=1, init=False, repr=False)
     max_seq_length: int = field(default=-1, init=False, repr=False)
-    train_dataset: Optional[torch.utils.data.Dataset] = field(default=None, init=False, repr=False)
-    test_dataset: Optional[torch.utils.data.Dataset] = field(default=None, init=False, repr=False)
+
+    def __post_init__(self) -> None:
+        # Could be a remote path (s3://) or a local path
```

Comment:
Do you want this dataset to support S3 paths?

Reply:
If it's straightforward, sure. But it's not a priority, this is mainly for debugging and testing